### PR TITLE
Feature/homework7-8

### PR DESCRIPTION
**Pull request by n-balashov**
Implementing integration with MovieDB

### CODE REVIEW

File: app/src/main/java/com/aacademy/homework/data/api/NetworkInterceptor.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.aacademy.homework.data.api
+
+import okhttp3.CacheControl
+import okhttp3.Interceptor
+import okhttp3.Response
+import java.util.concurrent.TimeUnit
+
+class NetworkInterceptor : Interceptor {
```
Comment (Pavel-Sidorovich) [inline app/src/main/java/com/aacademy/homework/data/api/NetworkInterceptor.kt]: Delete please

File: app/src/main/java/com/aacademy/homework/ui/movielist/FragmentMoviesList.kt [hunk 2]
```diff
@@ -65,31 +107,46 @@ class FragmentMoviesList : Fragment(R.layout.fragment_movies_list) {
                 setHasFixedSize(true)
                 adapter = movieAdapter
                 itemAnimator = MovieItemAnimator()
```
Comment (Pavel-Sidorovich) [inline app/src/main/java/com/aacademy/homework/ui/movielist/FragmentMoviesList.kt:109]: Please delete it breaks the UI.

File: app/src/main/java/com/aacademy/homework/ui/movielist/MovieAdapter.kt [hunk 3]
```diff
@@ -46,39 +37,40 @@ class MovieAdapter(
         )
 
     override fun onBindViewHolder(holder: MovieViewHolder, position: Int) {
-        holder.bind(moviePreviews[position])
+        holder.bind(movies[position])
     }
```
Comment (Pavel-Sidorovich) [inline app/src/main/java/com/aacademy/homework/ui/movielist/MovieAdapter.kt]: Maybe single line?

File: app/src/main/java/com/aacademy/homework/di/ApiModule.kt [hunk 4]
```diff
@@ -28,33 +28,41 @@ object ApiModule {
...
+            }
             .build()
     }
 
     @Singleton
     @Provides
-    fun provideConverterFactory(): Converter.Factory {
-        return Json { ignoreUnknownKeys = true }.asConverterFactory("application/json".toMediaType())
+    fun provideCache(@ApplicationContext context: Context): Cache {
+        val cacheSize = 5 * 1024 * 1024.toLong()
```
Comment (Pavel-Sidorovich) [inline app/src/main/java/com/aacademy/homework/di/ApiModule.kt]: Delete, please don't use this.

File: app/src/main/java/com/aacademy/homework/di/ApiModule.kt [hunk 5]
```diff
@@ -28,33 +28,41 @@ object ApiModule {
...
     }
 
     @Singleton
     @Provides
-    fun provideOkHttpClient(): OkHttpClient {
+    fun provideOkHttpClient(cache: Cache): OkHttpClient {
         return OkHttpClient.Builder()
-            .apply { addInterceptor(HttpLoggingInterceptor().setLevel(if (BuildConfig.DEBUG) Level.BODY else Level.NONE)) }
+            .apply {
+//                cache(cache)
```
Comment (Pavel-Sidorovich) [inline app/src/main/java/com/aacademy/homework/di/ApiModule.kt]: ;-)

File: app/src/main/java/com/aacademy/homework/di/ApiModule.kt [hunk 6]
```diff
@@ -28,33 +28,41 @@ object ApiModule {
...
+                addInterceptor(HttpLoggingInterceptor().setLevel(if (BuildConfig.DEBUG) Level.BODY else Level.NONE))
+                addInterceptor(ApiKeyQueryInterceptor())
+//                addNetworkInterceptor(NetworkInterceptor())
```
Comment (Pavel-Sidorovich) [inline app/src/main/java/com/aacademy/homework/di/ApiModule.kt]: ;-)

File: app/src/main/java/com/aacademy/homework/ui/movielist/FragmentMoviesList.kt (hunk 2 above)
Comment (n-balashov) [reply app/src/main/java/com/aacademy/homework/ui/movielist/FragmentMoviesList.kt:109]: only partly

File: app/src/main/java/com/aacademy/homework/data/local/dao/AppDatabase.kt [hunk 7]
```diff
@@ -2,16 +2,16 @@ package com.aacademy.homework.data.local.dao
...
-import com.aacademy.homework.data.model.MoviePreview
+import com.aacademy.homework.data.model.Actor
+import com.aacademy.homework.data.model.Movie
 
 @Database(
-    entities = [MoviePreview::class, MovieDetail::class],
+    entities = [Movie::class, Actor::class],
     exportSchema = false,
-    version = 4
+    version = 9
```
Comment (nikslushkin) [inline app/src/main/java/com/aacademy/homework/data/local/dao/AppDatabase.kt]: I recommend you to revert it to 1. If app was not released there is no reason to increment it. Ask your testers to delete app before update

File: app/src/main/java/com/aacademy/homework/foundations/glide/CustomImageLoader.kt [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+import com.aacademy.homework.data.preferences.MyPreference
+import com.bumptech.glide.load.Options
+import com.bumptech.glide.load.model.GlideUrl
+import com.bumptech.glide.load.model.ModelLoader
+import com.bumptech.glide.load.model.ModelLoaderFactory
+import com.bumptech.glide.load.model.MultiModelLoaderFactory
+import com.bumptech.glide.load.model.stream.BaseGlideUrlLoader
+import java.io.InputStream
+
+class CustomImageLoader(urlLoader: ModelLoader<GlideUrl?, InputStream?>?, private val baseUrl: MyPreference) :
```
Comment (nikslushkin) [inline app/src/main/java/com/aacademy/homework/foundations/glide/CustomImageLoader.kt]: url is not a preference

**Review by nikslushkin — APPROVED** [review]

**Pull request merged**